src/components/tx/TxHistory/history.js

```diff
@@ -7,31 +7,99 @@ import { List } from 'immutable';
 import Card from '../../../elements/Card';
 import Header from './Header';
 import TxList from './List';
+import TokenUnits from '../../../lib/tokenUnits';
 
 import styles from './history.scss';
 
-const TransactionsHistory = ({ transactions, accountId }) => {
-  return (
-    <Card>
-      <div className={ styles.container }>
-        <Header />
-        <TxList transactions={ transactions } accountId={ accountId }/>
-      </div>
-    </Card>
-  );
+const getFieldForFilter = (txFilter) => {
+  if (txFilter === 'IN') {
+    return 'to';
+  }
+  if (txFilter === 'OUT') {
+    return 'from';
+  }
 };
 
-TransactionsHistory.propTypes = {
-  transactions: PropTypes.object.isRequired,
-  accountId: PropTypes.string,
-};
+type Props = {
+  accountId: string,
+  transactions: any,
+  accounts: any
+}
+
+class TransactionsHistory extends React.Component<Props> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      txFilter: 'ALL',
+      displayedTransactions: this.props.transactions,
+    };
+  }
+  onSearchChange(e) {
+    const value = e.target.value;
+    const filteredTxes = this.props.transactions.filter((tx) => {
```

Comment:
i've been reading up on reselect. looking to refactor code like this to use that in the future.